ansible_builder/main.py

```diff
@@ -428,12 +430,18 @@ def prepare_build_stage_steps(self):
         ])
         return self.steps
 
-    def prepare_final_stage_steps(self):
+    def prepare_final_stage_steps(self, collections):
         self.steps.extend([
             "",
             "FROM $ANSIBLE_RUNNER_IMAGE"
             "",
         ])
+        if collections:
+            self.steps.extend([
+                "",
+                "LABEL collections='{}'".format(json.dumps(collections)),
```

Comment:
I would recommend using `com.ansible.execution-environment.collections` or `com.ansible.collections` as the label for this, as per the docker recommendations: https://docs.docker.com/config/labels-custom-metadata/#key-format-recommendations.

We'll be using these labels to determine if an image is an EE, so making sure that they don't collide with labels other tools might use is important. `collections` is a pretty generic term and it could be used by other software.

Some other metadata should definitely be included:
- schema version, if you have an official schema for EE images, or at the very least the ansible-builder version. That way we can handle cases appropriately if there are any known bugs or format changes between different versions of builder.
- ansible version. Users will want to know which version of ansible is being used in the EE to help identify bugs and resolve CVEs

Some metadata that would be really nice to have but aren't critical:
- collection source: this would allow us to like to the source or documentation for a collection
- other dependencies: knowing what python or other binary deps are in an EE could also be helpful for identifying bugs or security issues

If you don't want to add all of this as different labels, another potential solution would be us use one label such as `com.ansible.execution-environment.metadata` and put all of this information in a json blob there. That would have the added benefit making it so there are fewer values to check when parsing the container config.